### PR TITLE
Enable `PT012`

### LINKED ISSUE #9028: Enable rules in `flake8-pytest-style`
Enable the following rules in https://beta.ruff.rs/docs/rules/#flake8-pytest-style-pt:

- [x] PT001
- [x] PT006
- [x] PT007
- [x] PT009
- [x] PT010
- [ ] PT012
- [x] PT018
- [x] PT023
- [x] PT026

## Settings

```toml
[tool.ruff.flake8-pytest-style]
mark-parentheses = false
fixture-parentheses = false
```

### PR DESCRIPTION
**Pull request by EdAbati**
## Related Issues/PRs

Closes #9028

## What changes are proposed in this pull request?

Enables `ruff` rule `PT012`

FYI @harupy 

I am a bit unsure about my changes [here](https://github.com/mlflow/mlflow/pull/9158/files#diff-cc0b1dce628ed6e2946d88f1c390c19939efa2519b92a71c2d0c507fec86c508R175-R181) and [here](https://github.com/mlflow/mlflow/pull/9158/files#diff-7dcb6b8ab1317bbae2dd65ce42772caa7f2ae91cac3917ce7a64cf9c400c0d07R231-R239). Do we still have to test (with `pytest.raises`) exceptions that we artificially raise in the test itself? I turned these into `try-except` for now. Please let me know if I missed something :)

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@EdAbati Thanks for the PR! https://github.com/astral-sh/ruff/pull/6081 will be shipped soon in the next ruff release.

File: tests/tracking/test_tracking.py [hunk 1]
```diff
@@ -169,12 +169,16 @@ def test_start_run_context_manager():
...
+
     # Launch a separate run that fails, verify the run status is FAILED and the run UUID is
     # different
-    with pytest.raises(Exception, match="Failing run!"):
+    try:
         with start_run() as second_run:
             second_run_id = second_run.info.run_id
             raise Exception("Failing run!")
+    except Exception:
+        pass
```
Comment (harupy) [inline tests/tracking/test_tracking.py:180]: `with pytest.raises` is not equivalent to `try-catch`. The new code doesn't throw if `with start_run()` swallows the error, which is not the desired behavior here.

File: tests/autologging/test_autologging_client.py [hunk 2]
```diff
@@ -228,16 +228,16 @@ def test_client_correctly_operates_as_context_manager_for_synchronous_flush():
...
+    try:
         with mlflow.start_run(), MlflowAutologgingQueueingClient() as client:
             run_id_2 = mlflow.active_run().info.run_id
             client.log_params(run_id_2, params_to_log)
             client.log_metrics(run_id_2, metrics_to_log)
             client.set_tags(run_id_2, tags_to_log)
-            raise exc_to_raise
+            raise Exception("test exception")
+    except Exception:
+        pass
```
Comment (harupy) [inline tests/autologging/test_autologging_client.py:239]: We can disable `PT012` here.

File: pyproject.toml [hunk 3]
```diff
@@ -40,13 +39,13 @@ select = [
...
-  "F402",  # import-shadowed-by-loop-var
-  "F403",  # undefined-local-with-import-star
-  "F811",  # redefined-while-unused
+  "E402", # module-import-not-at-top-of-file
+  "E501", # line-too-long
+  "E741", # ambiguous-variable-name
+  "F401", # unused-import
+  "F402", # import-shadowed-by-loop-var
+  "F403", # undefined-local-with-import-star
+  "F811", # redefined-while-unused
```
Comment (harupy) [inline pyproject.toml:48]: Do we need this change?

Comment (EdAbati) [reply pyproject.toml:48]: I think it was changed by some automatic tooling I will revert

**Comment by harupy** [comment]
@EdAbati I filed #9200.

**Comment by harupy** [comment]
@EdAbati #9200 is merged, we can reset the changes in this PR, and re-run `ruff .` to see what needs to be fixed.

**Comment by harupy** [comment]
@EdAbati #9200 is merged.

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by EdAbati** [comment]
Thank you @harupy for finishing this PR! (I was away with no access to a computer) :)

**Pull request merged**